### PR TITLE
Add GetFileLink method to plugin API

### PR DESCRIPTION
**Pull request by Charliekenney23**
#### Summary
add GetFileLink method to plugin API.

#### Ticket Link
#9622

#### Checklist
[x] Added or updated uint tests (required for all new features)

### CODE REVIEW

File: plugin/api.go [hunk 1]
```diff
@@ -190,6 +190,9 @@ type API interface {
 	// GetFileInfo gets a File Info for a specific fileId
 	GetFileInfo(fileId string) (*model.FileInfo, *model.AppError)
 
+	// GetFileLink gets the public link to a file by fileId.
```
Comment (lieut-data) [inline plugin/api.go]: Can you document the minimum version required here?
```suggestion
	// GetFileLink gets the public link to a file by fileId.
	//
	// Minimum server version: 5.6
```

File: app/plugin_api.go [hunk 2]
```diff
@@ -323,6 +323,23 @@ func (api *PluginAPI) GetFileInfo(fileId string) (*model.FileInfo, *model.AppErr
 	return api.app.GetFileInfo(fileId)
 }
 
+func (api *PluginAPI) GetFileLink(fileId string) (string, *model.AppError) {
+	if !api.app.Config().FileSettings.EnablePublicLink {
+		return "", model.NewAppError("GetFileLink", "plugin_api.get_file_link.disabled.app_error", nil, "", http.StatusNotImplemented)
```
Comment (lieut-data) [inline app/plugin_api.go]: @jwilander, do we need an error message to accompany the new `plugin_api.get_file_link.disabled.app_error` and `plugin_api.get_file_link.no_post.app_error`? I can't quite recall our stance on translating these kind of error messages.

Comment (jwilander) [reply app/plugin_api.go]: Ideally we'd have an error message that's __not__ translated but I'm not sure AppError supports that at the moment. I think this falls under general improvements to AppError in the future

Comment (lieut-data) [reply app/plugin_api.go]: Makes sense. For now, @Charliekenney23 can you update our `i18n/en.json` to add messages for these error ids?

**Comment by lieut-data** [comment]
Thanks @charliekenney23!

**Pull request merged**